### ISSUE TITLE
Can't create ibundle for linux kernel

### ISSUE THREAD

**Issue by JuzzKidd** [drmikehenry/git-ibundle#2]
Firstly, thanks for creating such a useful tool! It's been working flawlessly with most repos I've been sync'ing however I've hit a problem with the linux kernel repo. This is what's happening:

```
$ git clone --mirror https://github.com/torvalds/linux.git
$ cd linux
$ git ibundle create --verbose ../linux.ibundle
error: the git_object of id '5dc01c595e6c6ec9ccda4f6f69c131c0dd945f8c' can not be successfully peeled into a commit (git_object_t=1).; class=Object (11); code=Peel (-19)
```

Any ideas? Something to do with the repo containing lightweight tags?

**Comment by drmikehenry** [comment]
Git tags are expected to point (directly or indirectly) to a "commit" object for "normal" use, and git-ibundle currently requires this to be the case.  I hadn't expected there to be a reason to create a non-standard tag that points to a different object type, but Git's machinery does allow for it (<https://stackoverflow.com/questions/61336211/can-a-lightweight-tag-point-to-a-tree-object-or-blob-object>),  and it turns out that the kernel repository takes advantage of this flexibility.

Normally a tag (e.g., `v3.0`) points to an object id (`02f8c6aee8df3cdc935e9bdd4f2d020306035dbe` in this case) of type `commit`:

```console
$ git show v3.0
tag v3.0
Tagger: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Jul 21 19:17:29 2011 -0700

Linux 3.0

w00t!
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2.0.16 (GNU/Linux)

iEYEABECAAYFAk4o3cgACgkQF3YsRnbiHLvw3gCfYIx8Sw/s3s4LNh8ij7Rb3ltt
9RsAoJgdZdHrRrWeB3G1q92FcJtbMHu9
=m0tn
-----END PGP SIGNATURE-----

commit 02f8c6aee8df3cdc935e9bdd4f2d020306035dbe (tag: v3.0)
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Jul 21 19:17:23 2011 -0700

...
```

But the object ID from the error message proves to be a `tree` object instead of a `commit`:

```console
$ git show 5dc01c595e6c6ec9ccda4f6f69c131c0dd945f8c

tag v2.6.11-tree

This is the 2.6.11 tree object.

NOTE! There's no commit for this, since it happened before I started with git.
Eventually we'll import some sort of history, and that should tie this tree
object up to a real commit. In the meantime, this acts as an anchor point for
doing diffs etc under git.
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.2.4 (GNU/Linux)

iD8DBQBCeV/eF3YsRnbiHLsRAl+SAKCVp8lVXwpUhMEvy8N5jVBd16UCmACeOtP6
KLMHist5yj0sw1E4hDTyQa0=
=/bIK
-----END PGP SIGNATURE-----

tree 5dc01c595e6c6ec9ccda4f6f69c131c0dd945f8c

COPYING
...
```

The output shows that tag `v2.6.11-tree` points to this `tree` object. The associated comment explains the rationale for this unusual usage.

It turns out there is a second tag (`v2.6.11`) pointing to this `tree` as well.

One work-around is to remove these tags before using `git-ibundle`:

```console
$ git tag -d v2.6.11 v2.6.11-tree
```

Now `git-ibundle` should work correctly (though without the ability to mirror those two tags); however, after the next `git fetch`, the tags will be re-fetched from the upstream.  Re-deleting the tags after each `git fetch` works around the current `git-ibundle` limitations in my testing.

I tried to use a negative refspec in Git's `config` file to exclude these tags automatically, but I couldn't make this work.  Here's what I tried:

```
[remote "origin"]
        url = https://github.com/torvalds/linux.git
        fetch = +refs/*:refs/*
        fetch = ^refs/tags/v2.6.11
        fetch = ^refs/tags/v2.6.11-tree
        mirror = true
```

I don't yet know how much change will be required in `git-ibundle` to eliminate the assumption that all tags must point to `commit` objects.  I'll leave this issue open until I have time to investigate more fully.

**Comment by JuzzKidd** [comment]
Very much appreciate your quick and thorough reply! I used your advice and removed the tags with:

```
git tag -d v2.6.11 v2.6.11-tree
```

It worked a treat and I'll just make sure to run that after doing a fetch. Thanks!

**Comment by drmikehenry** [comment]
This has been fixed in v0.2.2.  Tags and branches that have no associated commits no longer cause errors.  You should be able to use v0.2.2 to pick up incremental bundling where you left off, without the need to delete any Linux kernel tags.

**Comment by JuzzKidd** [comment]
Amazing! Thank you!
